cockatrice/src/dlg_settings.cpp

```diff
@@ -464,25 +464,154 @@ void UserInterfaceSettingsPage::retranslateUi()
-    generalGroupBox->setLayout(generalGrid);
-    
-    QVBoxLayout *mainLayout = new QVBoxLayout;
-    mainLayout->addWidget(generalGroupBox);
-    
-    setLayout(mainLayout);
+    QGridLayout *lpGeneralGrid = new QGridLayout;
+    QGridLayout *lpSpoilerGrid = new QGridLayout;
+
+    // Get all the update options and put them into the Combo Box
+    QMap<int, QString> lacSettingTimes = settingsCache->getDownloadSpoilerTimeIntervals();
+    bool lbCurrentIndexDefaulted = true;
+    int lnIndexToInsert = 0;
+    for (auto lIterator = lacSettingTimes.begin(); lIterator != lacSettingTimes.end(); lIterator++, lnIndexToInsert++)
+    {
+        msDownloadSpoilersTimeIntervalComboBox.insertItem(lnIndexToInsert, lIterator.value());
+
+        // If this is the saved value, we can mark it off it
+        if (lIterator.key() == settingsCache->getDownloadSpoilerTimeMinutes())
+        {
+            msDownloadSpoilersTimeIntervalComboBox.setCurrentIndex(lnIndexToInsert);
+            lbCurrentIndexDefaulted = false;
+        }
+    }
+
+    // Set the combo box's selected if we don't have a setting in the settingsCache
+    if (lbCurrentIndexDefaulted)
+    {
+        settingsCache->setDownloadSpoilerTimeMinutes(lacSettingTimes.keys().last()); // greatest value
+        msDownloadSpoilersTimeIntervalComboBox.setCurrentIndex(lacSettingTimes.size() - 1);
+    }
+
+    mcDownloadSpoilersCheckBox.setChecked(settingsCache->getDownloadSpoilersStatus());
+
+    mpSpoilerSavePathLineEdit = new QLineEdit(settingsCache->getSpoilerDatabasePath());
+    mpSpoilerSavePathLineEdit->setReadOnly(true);
+    mpSpoilerPathButton = new QPushButton("...");
+    connect(mpSpoilerPathButton, SIGNAL(clicked()), this, SLOT(spoilerPathButtonClicked()));
+
+    // Update the GUI depending on if the box is ticked or not
+    setSpoilersEnabled(mcDownloadSpoilersCheckBox.isChecked());
+
+    // Create the layout
+    lpSpoilerGrid->addWidget(&mcDownloadSpoilersCheckBox, 0, 0);
+    lpSpoilerGrid->addWidget(&msDownloadSpoilersLabel, 0, 1);
+    lpSpoilerGrid->addWidget(&msDownloadSpoilersTimeIntervalComboBox, 0, 2);
+    lpSpoilerGrid->addWidget(&mcSpoilerSaveLabel, 1, 0);
+    lpSpoilerGrid->addWidget(mpSpoilerSavePathLineEdit, 1, 1);
+    lpSpoilerGrid->addWidget(mpSpoilerPathButton, 1, 2);
+    lpSpoilerGrid->addWidget(&mcNextUpdateTimeToolTipLabel, 2, 0);
+    lpSpoilerGrid->addWidget(&mcNextUpdateTimeLabel, 2, 1);
+
+    lpSpoilerGrid->setColumnStretch(0, 2);
+    lpSpoilerGrid->setColumnStretch(1, 4);
+    lpSpoilerGrid->setColumnStretch(2, 1);
+
+    lpGeneralGrid->addWidget(&mcGeneralMessageLabel, 0, 0);
+
+    // On a change to the check box, hide/unhide the other fields
+    // On a change to the combo box, send a signal to the timer threads
+    connect(&mcDownloadSpoilersCheckBox, SIGNAL(stateChanged(int)), settingsCache, SLOT(setDownloadSpoilerStatus(int)));
+    connect(&mcDownloadSpoilersCheckBox, SIGNAL(stateChanged(int)), this, SLOT(setSpoilersEnabled(int)));
```

Comment:
You may want to bind the `toggled(bool)` signal instead to avoid the `static_cast<bool>` in the receiving code
(Also, the int returned by stateChanged has more values than just 0-unchecked 1-checked)